clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -5282,6 +5282,12 @@ void Clang::ConstructHostCompilerJob(Compilation &C, const JobAction &JA,
     HostCompileArgs.push_back(TCArgs.MakeArgString(*CWD));
   }
 
+  // Add /external:W0 for MSVC based compilation. We are using /external:I
+  // which requires /external:Wn upon usage. Use of /external:W0 disables
+  // warnings from headers that are included with /external:I.
+  if (IsMSVCHostCompiler)
+    HostCompileArgs.push_back("/external:W0");
```

Comment:
Is there a test added for this case?

Reply:
This addition is tested here:  https://github.com/intel/llvm/blob/c28ceb83add0ecce58c30c1c04f0a67b639a2fb5/clang/test/Driver/sycl-host-compiler-old-model.cpp#L19